app/app.go

```diff
@@ -108,15 +108,18 @@ func New(options ...Option) (*App, error) {
 
 	if utils.T == nil {
 		if err := utils.TranslationsPreInit(); err != nil {
+			app.Shutdown()
```

Comment:
Alternatively, you could name the return parameters:

```go
func New(options ...Option) (a *App, err error) {
```

Then above (after line 103), add this:

```go
defer func() {
    if err != nil {
        app.Shutdown()
    }
}()
```

Would be a little less error prone as we wouldn't need to remember to add these lines before all the return statements.

Reply:
Oh, neat trick.

I just pushed updated commits with this rewrite.